modules/core/src/v2/coins/sol.ts

```diff
@@ -178,6 +178,28 @@ export class Sol extends BaseCoin {
     params: ExplainTransactionOptions,
     callback?: NodeCallback<SolTransactionExplanation>
   ): Bluebird<SolTransactionExplanation> {
-    throw new MethodNotImplementedError('explainTransaction method not implemented');
+    const self = this;
+    return co<SolTransactionExplanation>(function* () {
+      const factory = accountLib.register(self.getChain(), accountLib.Sol.TransactionBuilderFactory);
+      let rebuiltTransaction;
+
+      try {
+        rebuiltTransaction = yield factory
```

Comment:
You hace here try and then an if to comparing the rebuiltTransaction.
If the try is success, could be the rebuilt transaction undefined for the next if?